crates/refunder/src/refund_service.rs

```diff
@@ -136,25 +143,42 @@ impl RefundService {
                             return None;
                         }
                     };
-                    let is_refunded: bool = order_owner == Some(INVALIDATED_OWNER);
-                    Some((eth_order_placement.uid, is_refunded))
+                    let refund_status = match order_owner {
+                        Some(bytes) if bytes == INVALIDATED_OWNER => RefundStatus::Refunded,
+                        Some(bytes) if bytes == H160::zero() => RefundStatus::Invalid,
+                        // any other owner
+                        _ => RefundStatus::NotYetRefunded,
+                    };
+                    Some((eth_order_placement.uid, refund_status))
                 }
             })
             .collect::<Vec<_>>();
 
         batch.execute_all(MAX_BATCH_SIZE).await;
         let uid_with_latest_refundablility = futures::future::join_all(futures).await;
-        type TupleWithRefundStatus = (Vec<(OrderUid, bool)>, Vec<(OrderUid, bool)>);
-        let (refunded_uids, to_be_refunded_uids): TupleWithRefundStatus =
+        type TupleWithRefundStatus = (Vec<(OrderUid, RefundStatus)>, Vec<(OrderUid, RefundStatus)>);
+        let (refunded_uids, uid_with_latest_refundablility): TupleWithRefundStatus =
```

Comment:
Instead of partitioning multiple types I think it would be cleaner to create the three vectors you want then iterate once and add the results to the correct vec.

Reply:
What you can do instead of looping is to use a `Itertools::group_by` on the refund status. Then you would get a `HashMap<RefundStatus, Vec<OrderUid>>`